AdyenActions/Actions/RedirectAction.swift

```diff
@@ -25,17 +40,39 @@ public struct RedirectAction: Decodable {
     /// - Parameters:
     ///   - url: The URL to which to redirect the user.
     ///   - paymentData: The server-generated payment data that should be submitted to the `/payments/details` endpoint.
-    ///   - nativeRedirectData: Native redirect data.
+    ///   - type: The redirect flow  used by the action. Defaults to `redirect`.
+    ///   - nativeRedirectData: Native redirect data. Defaults to `nil`.
     ///   - paymentMethodType: The type of the payment method.
     public init(
         url: URL,
         paymentData: String?,
+        type: RedirectType = .redirect,
         nativeRedirectData: String? = nil,
         paymentMethodType: String? = nil
     ) {
         self.url = url
         self.paymentData = paymentData
+        self.type = type
         self.nativeRedirectData = nativeRedirectData
         self.paymentMethodType = paymentMethodType
     }
+
+    public init(from decoder: any Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.url = try container.decode(URL.self, forKey: .url)
+        self.paymentData = try container.decodeIfPresent(String.self, forKey: .paymentData)
+        self.type = try container.decode(RedirectType.self, forKey: .type)
```

Comment:
are we sure the RedirectType will always be returned?

otherwise we could do:

```suggestion
        self.type = (try? container.decode(RedirectType.self, forKey: .type)) ?? FALLBACK_VALUE
```

Reply:
Is there any other way to check its optionality? https://docs.adyen.com/api-explorer/Checkout/71/post/payments#responses-200-action-CheckoutRedirectAction-type

Reply:
I double checked with the backend team. This property is always coming.